File: Javascript/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let area = 0;
    let left = 0;
    let right = height.length - 1;
    let temp;

    while (left < right) {
        if (height[left] > height[right]) {
            temp = height[right] * (right - left);
            right--;
        } else {
            temp = height[left] * (right - left);
            left++;
        }
        if (temp > area) {
            area = temp;
        }
    }
    return area;
};

File: Python/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        area = 0
        left = 0
        right = len(height) - 1
        while left < right:
            if height[left] > height[right]:
                temp = height[right] * (right - left)
                right -= 1
            else:
                temp = height[left] * (right - left)
                left += 1
            # print(height[right])
            if temp > area:
                area = temp

        return area
